src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -580,13 +594,117 @@ ApiGlobe.prototype.setRange = function(pRange /*, bool*/ ) {
 };
 
 /**
+ * { Displaces the central point to a specific amount of pixels from its current position.
+ *  The view flies to the desired coordinate, i.e.is not teleported instantly. Note : The results can be strange in some cases, if ever possible, when e.g.the camera looks horizontally or if the displaced center would not pick the ground once displaced.}
+ *
+ * @param      {<type>}  pVector  The vector
+ */
+ApiGlobe.prototype.pan = function(pVector) {
+
+    this.scene.currentControls().pan(pVector.x,pVector.y);
+
+};
+
+/**
  * Returns the actual zoom level. The level will always be between the [getMinZoomLevel(), getMaxZoomLevel()].
- * @constructor
- * @param {Id} id - The id of a layer.
+ *
+ * @return     {<type>}  The zoom level.
+ */
+ApiGlobe.prototype.getZoomLevel = function() {
+    return this.scene.getMap().getZoomLevel();
+};
+
+
+/**
+ * Gets the current zoom level, which is an index in the logical scales predefined for the application.
+ * The higher the level, the closer to the ground.
+ * The level is always in the [getMinZoomLevel(), getMaxZoomLevel()] range.
+ *
+ * @param      {number}  zoom    The zoom
  */
+ApiGlobe.prototype.setZoomLevel = function(zoom) {
+
+    zoom = Math.max(2,zoom);
+
+    let distance = this.scene.getMap().computeDistanceForZoomLevel(zoom,this.scene.currentCamera());
+
+    this.setRange(distance);
+
+};
+
+
+/**
+ * Return the current zoom scale at the central point of the view.
+ *
+ * @param      {number}  pitch   Screen pitch, in millimeters ; 0.28 by default
+ * @return     {number}  The zoom scale.
+ */
+ApiGlobe.prototype.getZoomScale = function(pitch) {
+    // TODO: Error div size height in Chrome
+    // Screen pitch, in millimeters
+    pitch = (pitch || 0.28)/1000;
+
+    let camera = this.scene.currentCamera();
+    let center = this.scene.currentControls().getCenter();
+    let rayon = center.length();
+    let range = center.distanceTo(camera.camera3D.position);
+    let distance = rayon + range;
+    let alpha = camera.FOV / 180 * Math.PI * 0.5;
+    let phi = Math.PI - Math.asin(distance / rayon * Math.sin(alpha));
+    let projection;
```

Comment:
Would you mind adding some comments about what is going on here?
It's not obvious, even if you're using sensible variable names.